### ISSUE TITLE
Center loss

### ISSUE THREAD

**Issue by ShownX** [tensorflow/tensorflow#4971]
It seems that center loss seems good in the recognition area to minize the variation of cluter. There is a caffe implementation of this loss. 
URL: https://github.com/ydwen/caffe-face/

I wrote some code but it seems not correct:
```Python
def center_loss(features, labels, name):
    with tf.name_scope(name=name):
       feat_dim = features.get_shape().as_list()[1]
       batch_size = features.get_shape().as_list()[0]
       centers = tf.get_variable('center', shape=[FLAGS.num_classes, feat_dim])
       # compute loss & update center
       loss = 0
       for i in range(batch_size):
           ft = features[i, :]
           idx = labels[i]
           c = tf.gather(centers, idx)
          loss += tf.reduce_mean(tf.square(tf.sub(ft, c)))
      loss /= batch_size
```
Is there can help me to understand better with Tensorflow, I can make a PR.

**Comment by jart** [comment]
This was tagged as a feature request but you seem to be asking for support. [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow) is a better venue for these types of questions. We rely on the community to offer support to our users.

**Comment by godfanmiao** [comment]
https://github.com/godfanmiao/MNIST_CNN_CENTERLOSS_TENSORFLOW
